web-client/src/app/services/wallet/wallet.service.ts

```diff
@@ -0,0 +1,81 @@
+import { Injectable } from '@angular/core';
+import { EnclaveService } from 'src/app/services/enclave/enclave.service';
+import { SessionStore } from 'src/app/stores/session/session.store';
+import { never } from 'src/helpers/helpers';
+import { AlgorandTransactionSigned } from 'src/schema/entities';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class WalletService {
+  constructor(
+    private sessionStore: SessionStore,
+    private enclaveService: EnclaveService
+  ) {}
+
+  async createWallet(name: string, pin: string) {
+    try {
+      const res = await this.enclaveService.createWallet({
+        owner_name: name,
+        auth_pin: pin,
+      });
+
+      if ('Created' in res) {
+        const { wallet_id: walletId } = res.Created;
+        this.sessionStore.update({ walletId, name });
+      } else if ('Failed' in res) {
+        this.sessionStore.setError(res);
+      } else {
+        never(res);
+      }
+    } catch (err) {
+      this.sessionStore.setError(err);
+    }
+  }
+
+  async openWallet(walletId: string, pin: string) {
+    try {
+      const res = await this.enclaveService.openWallet({
+        wallet_id: walletId,
```

Comment:
Maybe we should have consistency between the enclave service vars & wallet `(walletId & wallet_id)`

Reply:
The `snake_case` identifiers are what the MessagePack protocol uses: the arguments here are directly part of that.

We could add another layer with translation code that converts between the `snake_case` and `camelCase` conventions, but this could get complex: we'd either need to implement something that does dynamic conversions (which would be hard to get type-checked), or we'll need something that can automatically generate the right variants and conversions. (Or we could manually write and maintain them, but I wouldn't be keen for that, from a maintenance perspective.)

Using the protocol identifiers directly means there's just one thing to keep in sync for now, so pragmatically, I'm erring towards that.